spectree/plugins/falcon_plugin.py

```diff
@@ -271,12 +277,26 @@ async def request_validation(self, req, query, json, headers, cookies):
                     raise
                 media = None
             req.context.json = json.parse_obj(media)
+        if form:
+            try:
+                form_data = await req.get_media()
+            except self.FALCON_HTTP_ERROR as err:
+                if err.status not in self.FALCON_MEDIA_ERROR_CODE:
+                    raise
+                req.context.form = None
+            else:
+                res_data = {}
+                async for x in form_data:
+                    res_data[x.name] = x
+                    await x.data  # TODO - how to avoid this?
```

Comment:
@vytas7 here in async version I dont know how to avoid passing the object through Pydantic model without awaiting it here. Any advice appreciated.

Reply:
@yedpodtrzitko It depends what you are trying to achieve. Do you want to buffer whole files in the memory? Or do you want to spool them to temp files like some other frameworks do?\

Falcon even puts a cap on a maximum amount of data that can be referenced this way (`await part.get_data()`) in order to avoid surprises such as running out of memory.

Use `await part.stream.read()` to read the whole part as a bytestring, or `await part.stream.pipe(async_file)`, or read by chunks, and store the result somewhere.